src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java

```diff
@@ -307,13 +308,17 @@ public boolean delete(Path f, boolean recursive) throws IOException {
 
     if ((fileStatus != null && fileStatus.isDirectory()) || (fileStatus == null && prefixBased)) {
       LOG.trace("{} is directory, prefix based listing set to {}", f.toString(), prefixBased);
-      result = storageClient.list(hostNameScheme, f, false, prefixBased);
+      FileStatus[] listing = storageClient.list(hostNameScheme, f, false, prefixBased);
+      for (FileStatus fs : listing) {
+        if (filter.accept(fs.getPath())) {
+          result.add(fs);
+        }
+      }
     } else if (fileStatus != null) {
```

Comment:
@gilv Because I'm unsure how to tell if it's the default filter that's passed or a custom one.